### DISCUSSION TITLE
Welcome to UraniumUI Discussions!

### DISCUSSION THREAD

**Discussion by enisn** [enisn/UraniumUI#37] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by victorsigga** [comment]
Hello guys, I want to ask if your UI kit is cross platform too. Will all the implementations work fine in Android, iOS and UWP ?

**Reply by enisn** [reply]
Yes, implementation will work on each platform that MAUI Supports.

**Reply by victorsigga** [reply]
Thats very nice, adding you guys on favorites list

**Comment by MichaelSvejcar** [comment]
Hello, first thank you for developing this thing, looks really useful! I have few questions - regarding the datagrid, do you think it's possible to make it scrollable so the grid will scroll, but the header will stay visible all the time? Also, is it possible to put pictures or different dynamic elements in the cells?
And one more question regarding treeView - is it possible to make it drag&drop, so you could just select multiple files and drag them to your app? 
Thanks for your response!

**Reply by enisn** [reply]
Hi @MichaelSvejcar

### DataGrid
A sticky header feature isn't included in the roadmap but it's beneficial and I'll consider it for adding to the roadmap. Also, horizontal scrolling is another topic, They might be developed simultaneously. I have horizontal scrolling in the roadmap.

### TreeView
Unfortunately, drag & drop feature isn't available for treeview at he moment and there is no plan for v1.0. It might be planned for v2.0

**Reply by MichaelSvejcar** [reply]
Thank you for the response @enisn!
How complicated do you think these features will be to implement by myself?
Thanks!

**Reply by enisn** [reply]
I've planned it for v1.1
https://github.com/enisn/UraniumUI/issues/73

**Comment by PinkBlackEagle** [comment]
Hi eins,
what a great job you did! I love UranuimUI! I have a question. How can i add a Fontawesome icon programmatically?
Via XAML ok but i would also need it programmatically
thank you so much and really GREAT job!

Thanks in advance

**Reply by enisn** [reply]
The following document should help with it:
https://learn.microsoft.com/en-us/dotnet/maui/user-interface/fonts#display-font-icons

<img width="712" alt="image" src="https://user-images.githubusercontent.com/23705418/198344058-91afce15-e004-4b5b-bf43-284b420b9f96.png">

**Reply by PinkBlackEagle** [reply]
Hi Enis,
it works fine in that way.

Thank you very much!

**Comment by bcaceiro** [comment]
Hello @enisn ! Regarding TabView I have a question: Is it possible to generate new tabs on-demand ( or binding to a collection of tabs). As far as I read/understand, I can't achieve this.  I've tried to:  tabViewContainer.Add( new TabItem), but the TabItem is a bindable object, nnot a MAUI view. Any idea how to accomplish this?

**Reply by enisn** [reply]
You can name a TabView in XAML
```xml
<material:TabView x:Name="myTabView">
</material:TabView>
```

And you can add new tabs using this nem from **.cs** file.
```csharp
myTabView.Items.Add(new TabItem
{
	Title = "My Tab 1",
	Content = new Label { Text = "This is the content of my first tab" }
});
```

---

After my investigations, I saw Tabs aren't a bindable property. I've created a new issue to make the **Items** property as BindableProperty.
https://github.com/enisn/UraniumUI/issues/93

**Reply by enisn** [reply]
After development of #93 

Tabs will be bindable and you can use binding like this
```
<material:TabView Items="{Binding MyTabs}">
</material:TabView>
```

**Reply by enisn** [reply]
Dynamic Tabs Support is added with https://github.com/enisn/UraniumUI/pull/183
